### PR TITLE
Implement order staging

### PR DESCRIPTION
**Pull request by 0-yibai**

### CODE REVIEW

File: src/main/java/seedu/address/model/order/stage/SentForDeliveryState.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package seedu.address.model.order.stage;
+
+public class SentForDeliveryState implements StageState {
```
Comment (chonghaoen) [inline src/main/java/seedu/address/model/order/stage/SentForDeliveryState.java:3]: would sentfordeliverystate be too troublesome? maybe just under preparation, ready for delivery and received is enough?

**Review by chonghaoen — COMMENTED** [review]
nice implementation of stages :)

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -147,6 +147,13 @@ public Order editOrder(Order target, Product currProduct, Quantity newQuantity)
         return addressBook.editOrder(target, currProduct, newQuantity);
     }
 
+    @Override
+    public Order goToNextStage(Order target) {
+        requireNonNull(target);
+
+        return addressBook.goToNextStage(target);
```
Comment (nigel27022001) [inline src/main/java/seedu/address/model/ModelManager.java:154]: Good abstraction

File: src/main/java/seedu/address/model/order/Order.java [hunk 3]
```diff
@@ -251,15 +273,14 @@ public int hashCode() {
...
-            str += ",";
-            str += productMap.get(productList.get(k)).getValue();
-            str += "\n";
+        ArrayList<Product> productList = new ArrayList<>(set);
+        StringBuilder str = new StringBuilder();
+        for (Product product : productList) {
+            str.append(product.getName());
+            str.append(",");
+            str.append(productMap.get(product).getValue());
+            str.append("\n");
```
Comment (bertrandong) [inline src/main/java/seedu/address/model/order/Order.java:282]: Perhaps use of String.format() here would be better? For consideration.

**Review by bertrandong — COMMENTED** [review]
Overall, good implementation and code structure. LGTM

**Pull request merged**